Does not compile on latest nightly

I am getting this error:
```
error[E0308]: mismatched types                                                                                                                                                                                     
   --> /home/r/.cargo/registry/src/github.com-1ecc6299db9ec823/compiletest_rs-0.3.14/src/lib.rs:101:22                                                                                                             
    |                                                                                                                                                                                                              
101 |         run_ignored: config.run_ignored,                                                                                                                                                                     
    |                      ^^^^^^^^^^^^^^^^^^ expected enum `test::RunIgnored`, found bool                                                                                                                         
    |                                                                                                                                                                                                              
    = note: expected type `test::RunIgnored`                                                                                                                                                                       
               found type `bool`                                                                                                                                                                                   
                                                                                                                                                                                                                   
error: aborting due to previous error                                                                                                                                                                              
                                                                                                                                                                                                                   
For more information about this error, try `rustc --explain E0308`.                                                                                                                                                
```

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -95,10 +95,16 @@ pub fn run_tests(config: &Config) {
 }
 
 pub fn test_opts(config: &Config) -> test::TestOpts {
+    #[cfg(feature = "stable")]
+    let run_ignored = config.run_ignored;
+
+    #[cfg(not(feature = "stable"))]
+    let run_ignored = if config.run_ignored { test::RunIgnored::Yes } else { test::RunIgnored::No };
+
     test::TestOpts {
         filter: config.filter.clone(),
         filter_exact: config.filter_exact,
-        run_ignored: if config.run_ignored { test::RunIgnored::Yes } else { test::RunIgnored::No },
+        run_ignored: run_ignored,
         format: if config.quiet { test::OutputFormat::Terse } else { test::OutputFormat::Pretty },
         logfile: config.logfile.clone(),
         run_tests: true,

